includes/Core/Admin/Screen.php

```diff
@@ -129,9 +130,55 @@ public function add( Context $context ) {
 						$this->args['capability'],
 						$this->slug,
 						'',
-						$context->url( 'dist/assets/images/logo-g_white_small.png' )
+						'data:image/svg+xml;base64,' . Google_Icon::to_base64()
 					);
 					$menu_slug = $this->slug;
+
+					/**
+					 * An SVG icon file needs to be colored (filled) based on the theme color setting.
+					 *
+					 * This exists in js as wp.svgPainter() per:
+					 * https://github.com/WordPress/WordPress/blob/master/wp-admin/js/svg-painter.js
+					 *
+					 * The downside of the js approach is that we get a brief flash of an unstyled icon
+					 * until the JS runs.
+					 *
+					 * A user can pick a custom Admin Color Scheme, which is only available in admin_init
+					 * or later actions. add_menu_page runs on the admin_menu action, which precedes admin_init
+					 * per https://codex.wordpress.org/Plugin_API/Action_Reference
+					 *
+					 * WordPress provides some color schemes out of the box, but they can also be added via
+					 * wp_admin_css_color()
+					 *
+					 * Our workaround is to set the icon and subsequently replace it in current_screen, which is
+					 * what we do in the following action.
+					 */
+					add_action(
+						'current_screen',
+						function() {
+							global $menu, $_wp_admin_css_colors, $pagenow;
+							$color_scheme = get_user_option( 'admin_color' ) ?: 'fresh';
+
+							// If we're on one of the sitekit pages, use the 'current' color, otherwise use the 'base' color.
+							// @see wp_admin_css_color().
+							$color_key = false === strpos( get_current_screen()->id, 'googlesitekit' ) ? 'base' : 'current';
+
+							if ( empty( $_wp_admin_css_colors[ $color_scheme ]->icon_colors[ $color_key ] ) ) {
+								return;
+							}
+
+							$color = $_wp_admin_css_colors[ $color_scheme ]->icon_colors[ $color_key ];
+
+							foreach ( $menu as &$item ) {
```

Comment:
This should be fine, but since `current_screen` will run any time `set_current_screen` is called, we shouldn't assume `$menu` will always be an array. Let's add a guard at the top to simply bail if `$menu` isn't an array to be extra safe.

Reply:
Resolved in https://github.com/google/site-kit-wp/pull/2834/commits/373c8c0fab5be7835786b64c91c6d21694140a6b